### ISSUE TITLE
Error Creating a GameServices object

### ISSUE THREAD

**Issue by liszto** [playgameservices/play-games-plugin-for-unity#1033]
Hi,
I tried to update my game to the new play-games-plugin for unity but I get an issue since this moment.
I'm trying to isolate the source of the error and I have few ideas.

- The update of play-games-plugin-for-unity from 0.9.26 (or earlier) to 0.9.32
- The update of the google-play-services from 6.X to 8.4
- The stripped google-play-services that we introduced with this update to decrease the references number in the DEX.

During my tests, I found some things but nothing really obvious.
- If we didn't strip the google-play-services we can't build due to DEX issue.
To strip I used JarJar.jar that could be found here : https://code.google.com/archive/p/jarjar/
But maybe it create some issue with the newly jar created ?

- If I used the old play play-games-plugin-for-unity with stripped "google-play-services" I get the same issue (but it could be due to using a recent google-play-service with an old play-games-plugin-for-unity
The issue in logcat is this one : 

`03-08 14:56:32.933: E/GamesNativeSDK(8287): Can't register class com/google/android/gms/common/data/DataBufferUtils: an exception occurred.
03-08 14:56:32.940: E/GamesNativeSDK(8287): Can't register class com/google/android/gms/appstate/AppStateManager: an exception occurred.
03-08 14:56:32.966: E/GamesNativeSDK(8287): Could not register one or more required Java classes.
03-08 14:56:33.008: I/Unity(8287): InvalidOperationException: There was an error creating a GameServices object. Check for log errors from GamesNativeSDK
03-08 14:56:33.008: I/Unity(8287):   at GooglePlayGames.Native.PInvoke.GameServicesBuilder.Build (GooglePlayGames.Native.PInvoke.PlatformConfiguration configRef) [0x00000] in <filename unknown>:0 
03-08 14:56:33.008: I/Unity(8287):   at GooglePlayGames.Native.NativeClient.InitializeGameServices () [0x00000] in <filename unknown>:0 
03-08 14:56:33.008: I/Unity(8287):   at GooglePlayGames.Native.NativeClient.Authenticate (System.Action`1 callback, Boolean silent) [0x00000] in <filename unknown>:0 
03-08 14:56:33.008: I/Unity(8287):   at GooglePlayGames.PlayGamesPlatform.Authenticate (System.Action`1 callback, Boolean silent) [0x00000] in <filename unknown>:0 
03-08 14:56:33.008: I/Unity(8287):   at GooglePlayGames.PlayGamesPlatform.Authenticate (System.Action`1 callback) [0x00000] in <filename unknown>:0 
03-08 14:56:33.008: I/Unity(8287):   at GooglePlayGames.PlayGamesLocalUser.Authenticate (System.Action`1 callback) [0x00000] in <filename unknown>:0 `

So I've got a question what are the compulsory dependancies to login with Google Play Games ?

The stripping rules are this ones : 
keep com.google.ads.*
keep com.google.android.gms.ads.*
keep com.google.android.gms.auth.*
keep com.google.android.gms.games.*
keep com.google.android.gms.gcm.*
keep com.google.android.gms.identity.*
keep com.google.android.gms.nearby.*
keep com.google.android.gms.plus.*
keep com.google.android.gms.security.*
keep com.google.android.gms.wallet.*

**Comment by claywilkinson** [comment]
AppStateManager has been removed from the java client.  The Snapshot (Saved Games) API should be used instead.  If you are getting this error, you are most likely using an older version of the plugin (specifically the libgpg.so which is included) compared to the java client.  Upgrading the plugin should address this.

The new plugin (0.9.32) uses the split aars to resolve the client dependencies, which should keep you well below the DEX limit.  Please share your results if this is not the case.

**Comment by liszto** [comment]
So if I resume my situation : 
- Update google play plugin for unity to 0.9.32
- Update Google Play Services to 8.4.0 but for this step, I used the complete jar download with the sdk manager in extra folder but in fact I must use the ".aar" files present in extra folder ?
Then I must select only the "aar" I need then trying to rebuild ?
(Actually I use the complete google play library services only for some ads providers, google play games and in app purchase)
so I think I can only use : ads / games / base / basement / nearby / wallet ? Maybe some others can be usefull ?

**Comment by claywilkinson** [comment]
You can determine other aar files you might need be looking at the list of aars: https://developers.google.com/android/guides/setup#split

You should not have to add those files to this plugin (They should be there after you run setup) in Assets/Plugins/Android/MainLibProj.

**Comment by liszto** [comment]
I know but we create our own pipeline so we remove your setup from our project to have a total control on all the building process.
We did this for all our plugin.

So I changed a lot of thing in our Android pipe during the last few days.
Now all plugins files are dropped in Android{plugins_folder}/libs
We removed all AndroidManifest and project.properties from Android{plugins_folder} root.
And we let Unity do the job for the apk integration we just generate the final manifest in preprocess phase.
And it seems to be ok :)

It's simplier than before where we need to have a bin, libs, res folder now we tried to have only "aar" files when they are available for a plugin, if their not we tried to keep a "bin" and "res" hierarchy + AndroidManifest and project.properties to let unity know that we need to integrate this non-aar files in our final apk.

To explane it's probably quite a mess but if I could demo this it's in fact quite simple and well organized.
